### PR TITLE
docs: docs housekeeping

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:
- add changelog link to module info page
- add title to license page
- link changelog to releases page

### CODE REVIEW

**Comment by yordis** [comment]
Thank you so much! CI is failing btw it seems you need to run the formatter

File: mix.exs [hunk 1]
```diff
@@ -37,7 +37,10 @@ defmodule Tesla.Mixfile do
     [
       maintainers: ["Tymon Tobolski"],
       licenses: ["MIT"],
-      links: %{"GitHub" => "https://github.com/elixir-tesla/tesla"}
+      links: %{
+        "GitHub" => @source_url,
+        "Changelog" => "#{@source_url}/releases"
```
Comment (yordis) [inline mix.exs:42]: After https://github.com/elixir-tesla/tesla/pull/728 is merged, we will have a CHANGELOG file

File: mix.exs [hunk 2]
```diff
@@ -93,17 +96,14 @@ defmodule Tesla.Mixfile do
...
         "guides/howtos/migrations/v1-macro-migration.md"
       ],
       extra_section: "GUIDES",
       logo: "guides/elixir-tesla-logo.png",
       extras:
         [
           "README.md",
-          "LICENSE"
-          # TODO: add CHANGELOG.md
-          # "CHANGELOG.md"
```
Comment (yordis) [inline mix.exs:106]: we will need this changelog after https://github.com/elixir-tesla/tesla/pull/728

**Review by yordis — APPROVED** [review]

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Comment by yordis** [comment]
(correction, it is much, and deeply appreciated) you always remind me this meme

image

**Pull request merged**